source/music.lua

```diff
@@ -346,7 +357,8 @@ function music.loadStageMusicInDir(stageid, name)
 
 					-- Insert the newly loaded track into a random position in the playlist
 					local pos = math.random(1, #STAGE_TRACKS[stageid])
-					table.insert(STAGE_TRACKS[stageid], pos, source)
+					local prob = tonumber(string.sub(string.match(filepath, "[^%._\n]+(_%d+)%.%w+$") or "_1", 2))
```

Comment:
Any reason you're capturing and subbing out the underscore?

I think this could be slightly simplified to

```lua
local prob = tonumber(string.match("this/is/a/test_32.wav", "[^%._\n]+_(%d+)%.%w+$")) or 1
```

`tonumber` will return nil if it's not a valid number, so you can put the `or 1` after it instead of after the match.

Reply:
Done.